Cython/Compiler/Nodes.py

```diff
@@ -9737,8 +9741,10 @@ def _allocate_closure_temp(self, code, entry):
         self.seen_closure_vars.add(entry.cname)
         self.seen_closure_vars.add(cname)
 
-        self.modified_entries.append((entry, entry.cname))
-        code.putln("%s = %s;" % (cname, entry.cname))
+        self.modified_entries.append((entry, entry.cname, entry.force_not_declared_in_module_state))
+        entry_cname = code.entry_cname_in_module_state(entry)
+        entry.force_not_declared_in_module_state = True
+        code.putln(f"{cname} = {entry_cname};")
         entry.cname = cname
```

Comment:
Hmm, interesting. I wasn't aware of this 14 years old cname replacement hack. It's unfortunate, because as we see here, it now requires another hack to work around it.

What was the reason again why entries couldn't know their complete module state based cname themselves? That could avoid the need for a `force_not_declared_in_module_state` flag.